ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Tables/DeploymentVulnerabilitiesTable.tsx

```diff
@@ -0,0 +1,219 @@
+import { gql } from '@apollo/client';
+import { min } from 'date-fns';
+
+import LinkShim from 'Components/PatternFly/LinkShim';
+import useSet from 'hooks/useSet';
+import { getDistanceStrictAsPhrase } from 'utils/dateUtils';
+import { UseURLSortResult } from 'hooks/useURLSort';
+import { FixableIcon, NotFixableIcon } from 'Components/PatternFly/FixabilityIcons';
+import VulnerabilitySeverityIconText from 'Components/PatternFly/IconText/VulnerabilitySeverityIconText';
+import { VulnerabilitySeverity } from 'types/cve.proto';
+import { getEntityPagePath } from '../searchUtils';
+import { DynamicColumnIcon } from '../components/DynamicIcon';
+
+import EmptyTableResults from '../components/EmptyTableResults';
+import DeploymentComponentVulnerabilitiesTable, {
+    DeploymentComponentVulnerability,
+    ImageMetadataContext,
+    deploymentComponentVulnerabilitiesFragment,
+} from './DeploymentComponentVulnerabilitiesTable';
+import { getAnyVulnerabilityIsFixable, getHighestVulnerabilitySeverity } from './table.utils';
+
+export const deploymentWithVulnerabilitiesFragment = gql`
+    ${deploymentComponentVulnerabilitiesFragment}
+    fragment DeploymentWithVulnerabilities on Deployment {
+        id
+        images(query: $query) {
+            ...ImageMetadataContext
+        }
+        imageVulnerabilities(query: $query, pagination: $pagination) {
+            id
+            cve
+            summary
+            images(query: $query) {
+                imageId: id
+                imageComponents(query: $query) {
+                    ...DeploymentComponentVulnerabilities
+                }
+            }
+        }
+    }
+`;
+
+export type DeploymentWithVulnerabilities = {
+    id: string;
+    images: ImageMetadataContext[];
+    imageVulnerabilities: {
+        id: string;
+        cve: string;
+        summary: string;
+        images: {
+            imageId: string;
+            imageComponents: DeploymentComponentVulnerability[];
+        }[];
+    }[];
+};
+
+function formatVulnerabilityData(deployment: DeploymentWithVulnerabilities): {
+    id: string;
+    cve: string;
+    severity: VulnerabilitySeverity;
+    isFixable: boolean;
+    discoveredAtImage: Date | null;
+    summary: string;
+    affectedComponentsText: string;
+    images: {
+        imageMetadataContext: ImageMetadataContext;
+        componentVulnerabilities: DeploymentComponentVulnerability[];
+    }[];
+}[] {
+    const imageMap: Record<string, ImageMetadataContext> = {};
+    deployment.images.forEach((image) => {
+        imageMap[image.id] = image;
+    });
+
+    return deployment.imageVulnerabilities.map((vulnerability) => {
+        const { id, cve, summary, images } = vulnerability;
+        // Severity, Fixability, and Discovered date are all based on the aggregate value of all components
+        const allComponents = vulnerability.images.flatMap((img) => img.imageComponents);
+        const severity = getHighestVulnerabilitySeverity(allComponents);
```

Comment:
nit: can we make these variables a little more descriptive for better code legibility?